azavea/green-equity-demo
# Add animated spending arc to map

## Overview

Adds animated arcs illustrating the movement of spending from DC to each state using the [Leaflet.curve plugin library](https://github.com/elfalem/Leaflet.curve). Arc path animations begin when award spending at month is > 0. The duration of the animation is until the current month.
Closes #26 

### Demo

https://user-images.githubusercontent.com/36374797/225422046-1ebb9b22-4bf4-45a1-bd7b-0eea3e089d55.mov

### Notes

- Ideally the animated arc paths could "ease out" instead of abruptly stopping at the end. If we figure out how to do this we might want to keep the paths visible on the path and have them "complete" their path, however to have them up clutters the map so I removed them of once the animation completes.
- I moved the dynamic spending bucket to the bottom right corner instead of over DC. From what I can find, there's no easy way to overlay a dynamic React component over a lat,lng point (in the demo video above it's just hard-coded absolute positioning). I added it to a Leaflet marker using renderToStatic similar how we did for the person Icons, but this didn't update as the timeValue incremented up. I'll open this as a new card to explore (#76).

## Testing Instructions

- `scripts/update`
- `scripts/server`
- Play animation and confirm arc animations appear and spending bucket progress updates
- Confirm paths are removed once animation is complete
- Confirm can replay without any issues

 ## Checklist
